### PR TITLE
"Common good" vs "System" parachain clean up

### PR DESCRIPTION
**Pull request by bkontur**
## Summary 
The term "common good parachain" has been abandoned in favor of "system parachain" - e.g. [Joe's speech at Decoded2023](https://youtu.be/CSO-ERHK2gY?t=456). This pull request tries to fix and align code with this vision.

## Impact

The important change is implementation of `trait IsSystem` for `Id` [here](https://github.com/paritytech/polkadot-sdk/pull/1406/files#diff-0b7b4f5b962a18ce980354592b55ab2a27b5a2e9f6f8089ec803ca73853e8583R225-R229) where we changed condition from `< 1000` to `<= 1999`, which means that all parachain IDs bellow 1999 (included) are considered as "system parachain" IDs. This change has a direct impact on the following components:

#### [ChildSystemParachainAsSuperuser](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/origin_conversion.rs#L72-L88) 
This origin converter is used for allowing to process XCM `Transact` from "system parachain" on the relay chain - e.g. see [configuration for Kusama](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/runtime/kusama/src/xcm_config.rs#L92-L101). 
Only configured for Kusama, Westend, Rococo runtimes.

**No need for this feature anymore.** See [comment](https://github.com/paritytech/polkadot-sdk/pull/1406#issuecomment-1708218715).

#### [IsChildSystemParachain](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/barriers.rs#L310-L317)
`IsChildSystemParachain` is used with `AllowExplicitUnpaidExecutionFrom` barrier for checking XCM programs (they have to start with `UnpaidExecution` instruction).
Only configured for Kusama, Westend, Rococo runtimes.

**Overall the impact is low or mostly ok because it only allows unpaid execution for "system parachains" (e.g. AssetHub, BridgeHub...) on the relay chain.**

#### [SiblingSystemParachainAsSuperuser](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/origin_conversion.rs#L94-L114)

Not used anywhere in `polkadot-sdk` repo.

## Unresolved Questions
- [x] constants `LOWEST_USER_ID` and `LOWEST_PUBLIC_ID` seem to express the same thing now, do we want to keep them both or deprecated one of them? If so, which one?
- [x] determine impact for `ChildSystemParachainAsSuperuser`

## TODO

- [ ] when merged here, open PR to the `polkadot-fellows`

## Related Material
https://youtu.be/CSO-ERHK2gY?t=456
https://forum.polkadot.network/t/polkadot-protocol-and-common-good-parachains/866
https://wiki.polkadot.network/docs/learn-system-chains

### CODE REVIEW

**Review by gilescope — APPROVED** [review]

**Comment by al3mart** [comment]
> ??? Unknown impact ??? Do we have any scenario where "system parachain" sends Transact to the relay chain?

In ours published docs we point to opening HRMP channels using a Transact instruction
https://docs.substrate.io/tutorials/build-a-parachain/open-message-passing-channels/

**Review by joepetrowski — APPROVED** [review]

**Comment by joepetrowski** [comment]
> constants `LOWEST_USER_ID` and `LOWEST_PUBLIC_ID` seem to express the same thing now, do we want to keep them both or deprecated one of them? If so, which one?

IMO OK to combine, slight preference for `LOWEST_PUBLIC_ID`.

**Review by acatangiu — APPROVED** [review]

**Review by franciscoaguirre — APPROVED** [review]

**Comment by bkontur** [comment]
> > constants `LOWEST_USER_ID` and `LOWEST_PUBLIC_ID` seem to express the same thing now, do we want to keep them both or deprecated one of them? If so, which one?
> 
> IMO OK to combine, slight preference for `LOWEST_PUBLIC_ID`.

@paritytech/polkadot-review Can you please confirm that we can remove [`LOWEST_USER_ID`](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/parachain/src/primitives.rs#L206)? It is only re-exported in [polkadot/primitives/src/v5/mod.rs](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/primitives/src/v5/mod.rs#L47). Or what do you suggest?

**Comment by joepetrowski** [comment]
I am 99.9% sure this value is never used anywhere except the logic to allow Root privilege from a parachain (which we've removed here anyway).

File: polkadot/parachain/src/primitives.rs [hunk 1]
```diff
@@ -199,13 +199,14 @@ impl From<i32> for Id {
...
 }
 
-const USER_INDEX_START: u32 = 1000;
+// System parachain ID is considered `< 2000`.
+const SYSTEM_INDEX_END: u32 = 1999;
 const PUBLIC_INDEX_START: u32 = 2000;
 
 /// The ID of the first user (non-system) parachain.
-pub const LOWEST_USER_ID: Id = Id(USER_INDEX_START);
+pub const LOWEST_USER_ID: Id = Id(PUBLIC_INDEX_START);
```
Comment (ordian) [inline polkadot/parachain/src/primitives.rs:207]: I was not able to find any reference on github other than forks of Polkadot: 
https://github.com/search?q=LOWEST_USER_ID+language%3ARust&type=code&ref=advsearch
(let me know if you know how to filter out forks in the search query)

so it should be fine to remove

**Review by ordian — APPROVED** [review]

**Comment by kianenigma** [comment]
I expect this to also update the LOWEST_PUBLIC_ID in https://github.com/polkadot-fellows/runtimes/blob/main/relay/polkadot/src/lib.rs.

**Review by kianenigma — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-2-0/4451/1

**Pull request merged**